### PR TITLE
Fix bugs for v1.12

### LINKED ISSUE #4984: The stop words are ignored by Meili
It seems that Meili is ignoring the stop words I set. According to the [specification](https://specs.meilisearch.dev/specifications/text/0123-stop-words-setting-api.html#_3-functional-specification), stop words should be ignored in the search query during matching. However, it appears that Meili is either not handling this as expected, or I may have misunderstood the concept of stop words.

I use the following command to update the settings of the index (which contains about 2000 items):
```
curl -X PATCH "${SEARCH_SERVER_URL}/indexes/${INDEX_NAME}/settings/" -H  'Content-Type: application/json' -H "Authorization: Bearer ${SEARCH_SERVER_KEY}" --data-binary @${SETTINGS_FILE}
```

The content of ${SETTINGS_FILE} file is following:
```
{
  "rankingRules": [
    "words",
    "typo",
    "proximity",
    "attribute",
    "sort",
    "exactness",
    "released:desc",
    "rank:desc"
  ],
  "searchableAttributes": [
    "asset_name",
    "asset_id",
    "title",
    "description",
    "actors",
    "director",
    "program_type",
    "creation_date",
    "released",
    "year",
    "suggested_price",
    "genre",
    "series_name"
  ],
  "stopWords": [
    "the",
    "a",
    "an"
  ],
  "filterableAttributes": [
    "program_type",
    "season",
    "episode",
    "actors",
    "suggested_price",
    "rated",
    "language",
    "display_as_new",
    "display_as_last_chance",
    "year",
    "director"
  ],
  "sortableAttributes": [
    "title",
    "released"
  ],
  "typoTolerance": {
    "disableOnAttributes": [
      "asset_id",
      "genre"
    ]
  }
}
```

Example of JSON items for index:
```
    {
        "asset_name": "The Launch",
        "creation_date": "29 May 2020",
        "description": "Four-star general Mark Naird - leader of the newly created Space Force - is pressured to launch a satellite despite dire warnings from scientists.",
        "asset_class": "episode",
        "title": "The Launch",
        "genre": [
            "comedy"
        ],
        "director": "Paul King",
        "actors": [
            "Steve Carell",
            "John Malkovich",
            "Ben Schwartz"
        ],
        "season": 1,
        "episode": 1,
        "year": 2020,
        "rated": "TV-MA",
        "released": "29 May 2020",
        "language": "en",
        "runtime": "35 min",
        "imdbrating": 6.8,
        "imdbvote": "N/A",
        "licensing_window_start": "2023-08-06",
        "licensing_window_end": "2027-01-20",
        "preview_period": 30,
        "display_as_new": "N",
        "display_as_last_chance": "N",
        "audio_type": "AC3",
        "hdcontent": "Y",
        "maximum_viewing_length": "48:00",
        "series_name": "Space Force",
        "subtitle_languages": "en",
        "closed_captioning": "Y",
        "suggested_price": 0.0,
        "program_type": "episode"
    },
```

The settings, aside from stopWords, seem to be working correctly, as I can see their effect in the Meili Web UI. For example, if some field is not included in searchableAttributes, the search results do not return matching items for that field when a query is executed. Meili logs also do not contain any errors during the settings are applied.

As you can see, I have set "the," "a," and "an" as stop words. However, in the screenshot below, when the search query contains these words, results are still returned. I expect that when a stop word is set, Meili would exclude the corresponding items from the results during the search query execution (and avoid highlighting them too).

**Screenshots**
image

**Meilisearch version:**
v1.10.2

### PR DESCRIPTION
**Pull request by ManyTheFish**
# Pull Request

## Related issue
Fixes #4984
Fixes https://github.com/meilisearch/meilisearch/issues/4974
Fixes [SDK test](https://github.com/meilisearch/meilisearch/actions/runs/11886701996/job/33118278794)
## What does this PR do?
- add 3 tests
- fix bugs

### CODE REVIEW

File: crates/milli/src/update/index_documents/extract/extract_docid_word_positions.rs [hunk 1]
```diff
@@ -57,9 +57,9 @@ pub fn extract_docid_word_positions<R: io::Read + io::Seek>(
         .map(|s| s.iter().map(String::as_str).collect());
     let old_dictionary: Option<Vec<_>> =
         settings_diff.old.dictionary.as_ref().map(|s| s.iter().map(String::as_str).collect());
-    let del_builder =
+    let mut del_builder =
         tokenizer_builder(old_stop_words, old_separators.as_deref(), old_dictionary.as_deref());
-    let del_tokenizer = del_builder.into_tokenizer();
+    let del_tokenizer = del_builder.build();
```
Comment (Kerollmops) [inline crates/milli/src/update/index_documents/extract/extract_docid_word_positions.rs:62]: I'll only merge once an issue opens on Charabia.

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: crates/milli/src/update/index_documents/extract/extract_docid_word_positions.rs (hunk 1 above)
Comment (ManyTheFish) [reply crates/milli/src/update/index_documents/extract/extract_docid_word_positions.rs:62]: related to https://github.com/meilisearch/charabia/issues/319

**Review by Kerollmops — APPROVED** [review]
bors merge

**Comment by Kerollmops** [comment]
bors merge

**Comment by Kerollmops** [comment]
bors merge

**Comment by Kerollmops** [comment]
bors merge

**Comment by Kerollmops** [comment]
bors merge

**Pull request merged**